Add deprecation documentation for several small linker features (#31302)
dotnet/docs: docs/core/compatibility/deployment/7.0/deprecated-trimmer-default-action.md

---
title: "Breaking change: Deprecation of `TrimmerDefaultAction` property"
description: Learn about the .NET 7 breaking change in deployment where the `TrimmerDefaultAction` property is being deprecated.
ms.date: 09/20/2022
---
# MSBuild property `TrimmerDefaultAction` is deprecated

> The value of the property `TrimmerDefaultAction` is going to be ignored by the publish process.

Reviewer: ```suggestion
The value of the `TrimmerDefaultAction` property is now ignored by the publish process.
```

## Previous behavior

Previously, only assemblies that were opted-in with `<IsTrimmable>true</IsTrimmable>` in the library project file were trimmed with the action specified by the `TrimmerDefaultAction`. The default value for that property was `link` in .NET 6.

## New behavior

Starting in .NET 7, the property `TrimmerDefaultAction` is ignored and publishing behaves as if it was set to `link` all the time.

Since the default value for the property was already set to the most aggressive mode `link`, setting it explicitly in a project file would typically try to reduce the aggressiveness of trimming. The most common setting was `copyused`. This means the setting affected almost always apps which produced trim warnings.

> If your app did have trim warnings, you may see changes in behavior or exceptions.

Reviewer: ```suggestion
If your app had any trim warnings, you may see changes in behavior or exceptions.
```
